products.md

```diff
@@ -53,6 +53,150 @@ Product::create([
+
+```php
+$finder = new ProductSearch();
+$result = $finder->nameContains('Shiny Glue')->getResults();
+```
+
+#### Starts With:
+
+Finds products and master products whose names start with "Mature", such as "Matured Cheese" or "Mature People".
+
+```php
+$finder = new ProductSearch();
+$result = $finder->nameStartsWith('Mature')->getResults();
+```
+
+#### Ends With:
+
+Finds products and master products where the name ends with "Transformator", such as "Bobinated Transformator".
+
+```php
+$finder = new ProductSearch();
+$result = $finder->nameEndsWith('Transformator')->getResults();
+```
+
+#### Multiple Matches:
+
+Returns multiple products and master products containing "Mandarin" in their names.
+
+```php
+$finder = new ProductSearch();
+$products = $finder->nameContains('Mandarin')->getResults();
+```
+
+### Finding Products by Slug
+
+#### Exact Slug Match:
+
+Returns the product with the exact matching slug.
+
+```php
+$product = ProductSearch::findBySlug('nintendo-todd-20cm-plush');
+```
+
+#### Slug with Exception Handling:
+
+Throws a `ModelNotFoundException` if the product is not found.
+
+```php
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
+try {
+    $product = ProductSearch::findBySlugOrFail('non-existent-slug');
+} catch (ModelNotFoundException $e) {
+    // Handle case where product is not found
+}
+```
+
+### Filtering Products
+
+#### By Taxon:
+
+```php
+use Vanilo\Category\Models\Taxon;
+
+$taxon = Taxon::find(1);
+$products = (new ProductSearch())->withinTaxon($taxon)->getResults();
+```
+
+#### By Channel:
+
+```php
+use Vanilo\Channel\Models\Channel;
+
+$channel = Channel::find(1);
+$products = (new ProductSearch())->withinChannel($channel)->getResults();
+```
+
+#### By Price Range:
+
+```php
+$products = (new ProductSearch())->priceBetween(100, 500)->getResults();
+```
+
+### Filtering by Property Values
+
+#### By Single Property Value:
+
+Filters products that have a specific property value.
+
+```php
+use App\Models\PropertyValue;
+
+$propertyValue = PropertyValue::find(1);
+$products = (new ProductSearch())->havingPropertyValue($propertyValue)->getResults();
+```
+
+#### By Multiple Property Values:
+
+Filters products that match multiple property values.
+
+```php
+$propertyValues = PropertyValue::whereIn('id', [1, 2, 3])->get();
+$products = (new ProductSearch())->havingPropertyValues($propertyValues->toArray())->getResults();
+```
+
+#### By Property Name and Values:
+
+Filters products by a property name and its corresponding values.
+
+```php
+$products = (new ProductSearch())->havingPropertyValuesByName('color', ['red', 'blue'])->getResults();
+```
+
+#### Using OR Conditions for Property Values:
+
+Applies an OR condition when filtering by multiple property values.
+
+```php
+$products = (new ProductSearch())->orHavingPropertyValues($propertyValues->toArray())->getResults();
+```
+
+### Filtering by Custom Attributes
+
```

Comment:
You must mention here that these custom attributes are normal fields of the products, that they must be added in separate migrations, and that they have nothing to do with the "properties" feature.